### PR TITLE
Adapted Maven to be more direct and work cleanly with IntelliJ

### PR DESCRIPTION
**Pull request by Paul19988**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -148,4 +104,58 @@
...
+                        </goals>
+                        <configuration>
+                            <artifactSet>
+                                <includes>
+                                    <include>*</include>
+                                </includes>
+                            </artifactSet>
+                            <relocations>
+                                <relocation>
+                                    <pattern>org.apache.commons.pool2</pattern>
```
Comment (HSGamer) [inline pom.xml:151]: Why do you shade this?

File: pom.xml [hunk 2]
```diff
@@ -5,61 +5,17 @@
...
-                        <phase>package</phase>
-                        <goals>
-                            <goal>shade</goal>
-                        </goals>
-                        <configuration>
-                            <createDependencyReducedPom>false</createDependencyReducedPom>
-                            <minimizeJar>false</minimizeJar>
-                            <relocations>
-                                <relocation>
-                                    <pattern>org.bstats.bukkit</pattern>
```
Comment (HSGamer) [inline pom.xml:46]: I think you should keep this.

File: pom.xml [hunk 3]
```diff
@@ -148,4 +104,58 @@
...
+        </resources>
+
+        <finalName>${project.name}-${project.version}</finalName>
+        <defaultGoal>clean install</defaultGoal>
+        <!-- Plugins -->
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.6.1</version>
```
Comment (HSGamer) [inline pom.xml:127]: Can go up to 3.7.0

File: pom.xml [hunk 4]
```diff
@@ -148,4 +104,58 @@
...
+                <configuration>
+                    <source>1.8</source>
+                    <target>1.8</target>
```
Comment (HSGamer) [inline pom.xml:130]: Can use `${java.version}`

File: pom.xml [hunk 5]
```diff
@@ -148,4 +104,58 @@
...
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <version>2.3</version>
```
Comment (HSGamer) [inline pom.xml:136]: 2.3 is a bit old. Can go up to 3.2.4

File: pom.xml (hunk 1 above)
Comment (Paul19988) [reply pom.xml:151]: Has always been part of my pom.xml files never really had a need to remove it

File: pom.xml (hunk 2 above)
Comment (Paul19988) [reply pom.xml:46]: I've gone ahead and reimplemented bstats, will be creating a PR shortly

File: pom.xml (hunk 3 above)
Comment (Paul19988) [reply pom.xml:127]: Updated, will be creating a PR shortly

File: pom.xml (hunk 4 above)
Comment (Paul19988) [reply pom.xml:130]: Purposely enforced 1.8 as that is what is currently widely used for Servers (Some that intend to compile may not be using 1.8 as default)

File: pom.xml (hunk 5 above)
Comment (Paul19988) [reply pom.xml:136]: Updated, will be creating a PR shortly

**Pull request merged**